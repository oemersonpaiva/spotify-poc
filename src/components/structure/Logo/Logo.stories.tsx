import { Meta, Story } from '@storybook/react/types-6-0'
import Logo from './Logo'

export default {
  title: 'structure/Logo',
  component: Logo
} as Meta

export const Default: Story = (args) => <Logo {...args} />
