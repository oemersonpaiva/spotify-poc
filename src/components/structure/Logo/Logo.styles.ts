import styled, { css } from 'styled-components'

export const Wrapper = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.black[200]};
    display: inline-flex;
    align-items: center;
    text-decoration: none;
    font-weight: ${theme.font.bold};
    transition: color 0.2s linear;

    &:hover {
      color: ${theme.colors.primary};
    }

    svg {
      fill: currentColor;
      height: 2.4rem;
      width: 2.4rem;
      margin-right: ${theme.spacings.xxsmall};
    }
  `}
`
