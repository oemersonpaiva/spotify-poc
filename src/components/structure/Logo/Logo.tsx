import { ReactComponent as LogoIcon } from 'assets/svgs/logo.svg'
import * as S from './Logo.styles'

const Logo = () => (
  <S.Wrapper href="/">
    <LogoIcon aria-label="Spotify POC" /> Spotify POC
  </S.Wrapper>
)

export default Logo
