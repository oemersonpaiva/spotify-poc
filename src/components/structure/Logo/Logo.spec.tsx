import { renderWithProviders } from 'utils/tests/helpers'
import Logo from './Logo'

describe('<Logo />', () => {
  it('should render the heading', () => {
    const { getByLabelText } = renderWithProviders(<Logo />)

    expect(getByLabelText(/spotify poc/i)).toBeInTheDocument()
    expect(getByLabelText(/spotify poc/i)).toMatchSnapshot()
  })
})
