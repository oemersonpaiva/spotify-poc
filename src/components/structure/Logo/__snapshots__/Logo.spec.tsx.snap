// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the heading 1`] = `
<svg
  aria-label="Spotify POC"
>
  logo.svg
</svg>
`;
